azure-pipelines.yml

```diff
@@ -2,72 +2,97 @@ trigger:
+        checkLatest: true
 
-steps:
-  - task: NodeTool@0
-    inputs:
-      versionSpec: '14.x'
-      checkLatest: true
+    - script: |
+          npm install yo
+          npm install 
+      displayName: 'npm install'
+      
+    - script: |
+          npm run lint 
+      displayName: 'npm lint'
 
-  - script: |
-        npm install yo
-        npm install 
-    displayName: 'npm install'
-    
-  - script: |
-        npm run lint 
-    displayName: 'npm lint'
+    - task: Npm@1
+      displayName: "npm Packaging"
+      inputs:
+          command: 'custom'
+          customCommand: 'pack'
 
-  - task: Npm@1
-    displayName: "npm Packaging"
-    inputs:
-        command: 'custom'
-        customCommand: 'pack'
+    - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
+      displayName: 'SBOM Generation Task'
+      inputs:
+        BuildDropPath: '$(Build.ArtifactStagingDirectory)'
 
-  - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
-    displayName: 'SBOM Generation Task'
-    inputs:
-      BuildDropPath: '$(Build.ArtifactStagingDirectory)'
+    - task: Bash@3
+      inputs:
+        targetType: 'inline'
+        script: |
+          if [[ $(Build.SourceBranch) =~ ^refs/tags/v?[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+          echo "##vso[task.setvariable variable=NPM_PUBLISH_FLAG;]true"
+          fi
 
-  - task: CopyFiles@2
-    inputs:
-      SourceFolder: '$(Build.SourcesDirectory)'
-      Contents: '*.tgz'
-      TargetFolder: '$(Build.ArtifactStagingDirectory)/Package'
-  - task: PublishBuildArtifacts@1
-    inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)/Package'
-      ArtifactName: 'drop'
-      publishLocation: 'Container'
+    - task: Bash@3
+      condition: and(succeeded(), eq(variables['NPM_PUBLISH_FLAG'], 'true'))
+      inputs:
+        targetType: 'inline'
+        script: |
+          VERSION=$(node -p -e "require('./package.json').version")
+          echo "##vso[task.setvariable variable=PACKAGEVERSION;]$VERSION"
 
-  - task: Npm@1
-    displayName: "npm Test"
-    inputs:
-      command: 'custom'
-      customCommand: 'test'
-    
-  - task: GitHubRelease@1
-    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
-    inputs:
-      gitHubConnection: 'github.com_marianan'
-      repositoryName: '$(Build.Repository.Name)'
-      action: 'create'
-      target: '$(Build.SourceVersion)'
-      tagSource: 'userSpecifiedTag'
-      tag: '$(Build.SourceBranch)'
-      releaseNotesSource: 'inline'
-      assets: '$(Build.ArtifactStagingDirectory)/Package/*'
-      addChangeLog: false
+    - task: PublishBuildArtifacts@1
+      inputs:
+        PathtoPublish: '$(Build.ArtifactStagingDirectory)'
+        ArtifactName: 'drop'
+        publishLocation: 'Container'
+
+    - task: Npm@1
+      displayName: "npm Test"
+      inputs:
+        command: 'custom'
+        customCommand: 'test'
```

Comment:
Testing should happen before any release, maybe l.33?

Reply:
Yes, I have moved that.